mongo/integration/client_test.go

```diff
@@ -792,6 +793,124 @@ func TestClient(t *testing.T) {
 	})
 }
 
+func TestClient_BSONOptions(t *testing.T) {
```

Comment:
What are your thoughts on running the test in parallel? I don't think there would be any conflict, though we couldn't run the subtests in parallel.

Reply:
Seems safe, will try it out.